src/components/swap/PendingModalContent.tsx

```diff
@@ -70,82 +145,110 @@ interface PendingModalStep {
 }
 
-function getContent(
-  step: PendingConfirmModalState,
-  approvalCurrency: Currency | undefined,
-  confirmed: boolean,
-  theme: DefaultTheme
-): PendingModalStep {
+function getContent(args: ContentArgs): PendingModalStep {
+  const { chainId, step, approvalCurrency, swapConfirmed, swapPending, tokenApprovalPending, theme, trade } = args
   switch (step) {
     case ConfirmModalState.APPROVING_TOKEN:
       return {
-        title: t`Approve permit`,
-        subtitle: t`Proceed in wallet`,
-        label: t`Why are permits required?`,
-        tooltipText: t`Permit2 allows token approvals to be shared and managed across different applications.`,
-        logo: <CurrencyLoader currency={approvalCurrency} />,
+        title: t`Allow trading ${approvalCurrency?.symbol} on Uniswap`,
+        subtitle: (
+          <>
+            <Trans>First, we need your permission to use your DAI for swapping.</Trans>{' '}
+            <ExternalLink href="https://support.uniswap.org/hc/en-us/articles/8120520483085">
+              <Trans>Why is this required?</Trans>
+            </ExternalLink>
+          </>
+        ),
+        label: tokenApprovalPending ? t`Pending...` : t`Proceed in your wallet`,
+        logo: <PaperIcon currency={approvalCurrency} loading={tokenApprovalPending} />,
       }
     case ConfirmModalState.PERMITTING:
       return {
-        title: t`Approve ${approvalCurrency?.symbol}`,
-        subtitle: t`Proceed in wallet`,
-        label: t`Why are approvals required?`,
-        tooltipText: t`This provides the Uniswap protocol access to your token for trading. For security, this will expire after 30 days.`,
+        title: t`Unlock ${approvalCurrency?.symbol} for swapping`,
+        subtitle: (
+          <>
+            <Trans>This will expire after 30 days for your security.</Trans>{' '}
+            <ExternalLink href="https://support.uniswap.org/hc/en-us/articles/360056642192">
+              <Trans>Why is this required?</Trans>
+            </ExternalLink>
+          </>
+        ),
+        label: t`Proceed in your wallet`,
         logo: <CurrencyLoader currency={approvalCurrency} />,
       }
     case ConfirmModalState.PENDING_CONFIRMATION:
       return {
-        title: t`Confirm Swap`,
-        subtitle: t`Proceed in wallet`,
-        logo: confirmed ? <SizedAnimatedConfirmation /> : <Loader stroke={theme.textTertiary} size="48px" />,
+        title: swapPending ? t`Transaction submitted` : swapConfirmed ? t`Success` : t`Confirm Swap`,
+        subtitle: swapConfirmed ? (
+          <ExternalLink href={`https://etherscan.io/tx/${swapConfirmed}`} color="textSecondary">
+            <Trans>View on Explorer</Trans>
+          </ExternalLink>
+        ) : trade ? (
+          <TradeSummary trade={trade} />
+        ) : null,
+        label: !swapPending && !swapConfirmed ? t`Proceed in your wallet` : null,
+        logo:
+          swapConfirmed || (swapPending && chainId === SupportedChainId.MAINNET) ? (
```

Comment:
the flow is different on mainnet because the swap tx takes longer to confirm. on mainnet, we just show the success icon once the tx is sent.

added a comment explaining this